sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/directconnectivity/StoreReader.java

```diff
@@ -158,13 +158,13 @@ public StoreReader(
                                 return Flux.error(e);
                             }
                         }
-                ).onErrorResume(t -> {
-
+                ).onErrorResume(throwable -> {
+                    Throwable unwrappedException = Exceptions.unwrap(throwable);
```

Comment:
some places we do unwrap and some places we don't. why

Reply:
Because of the way these exceptions are thrown from the above layer.